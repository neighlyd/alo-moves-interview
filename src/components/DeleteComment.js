import React from 'react';
import Modal from 'react-modal';

const DeleteComment = (props) => {

  return (
    <Modal
      isOpen={!!props.modalOpen}
      className="modal"
      closeTimeoutMS={200}
      contentLabel={props.formTitle}
      onRequestClose={props.handleModalClose}
    >
    <div>
      <h3>Delete This Comment?</h3>
      <button onClick={props.handleSubmit} className="delete__confirm_btn">Yes</button>
      <button onClick={props.handleModalClose} className="delete__deny_btn">No</button>
    </div>
    </Modal>
      
  )
}

export default DeleteComment;