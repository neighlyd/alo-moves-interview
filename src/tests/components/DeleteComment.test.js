import React from 'react';
import { shallow } from 'enzyme';

import DeleteComment from '../../components/DeleteComment';

let handleSubmit, wrapper;

beforeEach(() => {
  handleSubmit = jest.fn();
  wrapper = shallow(<DeleteComment handleSubmit={handleSubmit}/>)
});

it('DeleteComment should render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('DeleteComment should handleSubmit correctly', () => {
  wrapper.find('.confirm_btn').simulate('click');
  expect(handleSubmit).toHaveBeenCalled();
});