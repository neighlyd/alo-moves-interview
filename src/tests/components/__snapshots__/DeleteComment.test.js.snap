// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteComment should render correctly 1`] = `ShallowWrapper {}`;
